react-state/my-app/src/ToggleButton.js

```diff
@@ -0,0 +1,29 @@
+import { useState } from 'react';
+
+export default function ToggleButton({ text, color }) {
+  const [isClicked, setIsClicked] = useState(false);
+  console.log('useState: ', isClicked);
+  function handleClick() {
+    console.log('before setter: ', isClicked);
+    if (isClicked) {
+      setIsClicked(false);
+    } else {
+      setIsClicked(true);
+    }
+    console.log('after setter: ', isClicked);
+  }
+
+  if (isClicked) {
+    return (
+      <button onClick={handleClick} style={{ backgroundColor: color }}>
+        {text}
+      </button>
+    );
+  } else {
+    return (
+      <button onClick={handleClick} style={{ backgroundColor: 'white' }}>
+        {text}
+      </button>
```

Comment:
Can simplify this logic by using a ternary in `style`.

`style={{ backgroundColor: isClicked ? color : 'white' }}`